### PR TITLE
[CodeCompletion] Revert "Provide known module source file information"

### PR DESCRIPTION
**Pull request by rintaro**
This effectively reverts commit f8751d466e3ef7176113445d07fed61929f06c42.

We don't have plan to use this at this point.

### CODE REVIEW

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Comment by rintaro** [comment]
@swift-ci Please smoke test

File: lib/IDE/CodeCompletionCache.cpp [hunk 1]
```diff
@@ -209,7 +209,6 @@ static bool readCachedModule(llvm::MemoryBuffer *in,
     auto numBytesToErase = static_cast<unsigned>(*cursor++);
     auto chunkIndex = read32le(cursor);
     auto moduleIndex = read32le(cursor);
-    auto sourceFilePathIndex = read32le(cursor);
```
Comment (bnbarham) [inline lib/IDE/CodeCompletionCache.cpp:212]: `onDiskCompletionCacheVersion` wasn't bumped for the original change, but should it have been? In which case, should it be bumped now?

Comment (benlangmuir) [reply lib/IDE/CodeCompletionCache.cpp:212]: Heh, given that the version is "2" I suspect we've forgotten to update it multiple times now.  Yes, it should probably be bumped.

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Review by benlangmuir — APPROVED** [review]

File: lib/IDE/CodeCompletionCache.cpp (hunk 1 above)
Comment (rintaro) [reply lib/IDE/CodeCompletionCache.cpp:212]: Thanks for noticing! I bumped it to "3"

**Comment by rintaro** [comment]
@swift-ci Please smoke test

**Pull request merged**